staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap_test.go

```diff
@@ -414,8 +458,11 @@ func TestCreateConfigMap(t *testing.T) {
 					defer teardown()
 				}
 			}
+			err := configMapOptions.Validate()
```

Comment:
Why is this added?

Reply:
I was the one that remove the generator of the ConfigMap (https://github.com/pulls?q=is%3Apr+author%3Alauchokyip+archived%3Afalse+is%3Aclosed) and when I wrote the test cases for example this test cases https://github.com/kubernetes/kubernetes/blob/a45aeb626c7f2303c49466ae52833cd410cf88f2/staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap_test.go#L386 `Validate()` would catch the error here https://github.com/kubernetes/kubernetes/blob/a45aeb626c7f2303c49466ae52833cd410cf88f2/staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap.go#L208, same as what I did for `create_secret_test.go` https://github.com/kubernetes/kubernetes/blob/a45aeb626c7f2303c49466ae52833cd410cf88f2/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go#L480

Reply:
What I still don't understand is why this PR makes this change necessary. If the errors the cases you point to are meant to hit are in `Validate` and only `createConfigMap` is being called on master today, then how are the tests passing on master? Are they hitting errors other than the ones they're meant to? Incidentally, that possibility is one reason why I personally prefer to make error expectations strings to match on rather than booleans, to provide better assurances that the error that happens is the one I expect.

To be clear, I'm not opposed to this change--just want to make sure I understand it.

Reply:
Will take a look into it

Reply:
Turn out the error happened during `createConfigMap` 
For instance the test case `create_invalid_configmap_too_many_args`, instead of throwing the error `from-env-file cannot be combined with from-file or from-literal` it actually tries to create a config map from the file and it cannot read the file therefore throws an error

I agree comparing string error is way better so I will make a change from that

Reply:
Thanks for digging into this, and great job with the improvements to this file!

Reply:
@lauchokyip your latest rebase seems to have lost the changes from this discussion, i.e. the ones visible in https://github.com/kubernetes/kubernetes/compare/cc37d5df54caaaa50291d37c93042d73ea506778..977ca6feb4fe433e667df051dfa4b4c21e4fce02. Can you please restore them? Note that one of the error strings that shows up in that diff answers @eddiezane question. 😄

Reply:
Yes I am restoring them! Thanks for the reminder 😃